### ISSUE TITLE
Auto-link URLs

### ISSUE THREAD

**Issue by korelstar** [yydcdut/RxMarkdown#18]
It would be nice, if for any URL, a link is created automatically.

For `RxMDEditText`, every URL should be auto-linked (regardless if it is inside of a `[]()` or not). However, there will be a conflict between positioning the cursor by clicking and open the link by clicking. Therefore, I would propose to open the link by long-clicking (maybe using an additional menu), so that normal clicking still performs cursor positioning.

For `RxMDTextView`, links outside of a `[]()` should be linked. Here, opening the link can be called by normal clicking.

**Comment by yydcdut** [comment]
Thanks for your suggestions, I will do it these days!

**Comment by korelstar** [comment]
Currently, I'm using this regular expression to replace URLs in the text before assigning the text to RxMarkdown's `RxMDTextView`:

```
content = content.replaceAll("(?<![(])(https?://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|])(?![^\\[]*\\])", "[$1]($1)");
```

Feel free to adopt this directly in RxMarkdown or make something similar. 😃

**Comment by yydcdut** [comment]
Thank you!

**Comment by stefan-niedermann** [comment]
I think this has been implemented already in v0.1.2-alpha?

**Comment by yydcdut** [comment]
no, but next version!! Next version I will add it!!
